### ISSUE TITLE
Feature Request: Improve Alignment of Comments

### ISSUE THREAD

**Issue by Glitchy-Tozier** [rust-lang/rust-analyzer#11907]
This is the scenario I'm suggesting to improve. In the actual code these comments are aligned:

![rust_analyzer_comments](https://user-images.githubusercontent.com/59611881/161827087-64967f77-7b02-47c2-a6ae-012537841d50.png)

`cargo fmt` (at least I assume this is what is responsible) seems to be smart enough to automatically add spaces so that comments are well aligned.
It would be neat for rust-analyzer to also employ some mechanism with which it can (only visually) offset comments, when adding an (visual) offset to other parts of "comment-chains".

**Comment by lnicola** [comment]
I assume the offset is caused by the inlay hints? I don't think we can (actually, want to) do anything about this.

**Comment by Glitchy-Tozier** [comment]
Exactly, it was the hints. My thought was that there could also be temporary comment-offsets.

**Comment by lnicola** [comment]
The only thing we could do here is adding an empty inlay hint with the correct number of spaces to make the comments align. But that would be shown as a rectangle with the default styling.

Maybe the others think differently, but I still don't think we want to add this.

**Comment by Glitchy-Tozier** [comment]
> The only thing we could do here is adding an empty inlay hint with the correct number of spaces to make the comments align.

I would be happy with that.

> Maybe the others think differently, but I still don't think we want to add this.

Is there someone we could @ to get additional thoughts on the topic?

**Comment by lnicola** [comment]
@rust-analyzer/review ^

**Comment by Veykril** [comment]
Agree, this is not something the server should have to deal with imo
